test/functional/autocmd/dirchanged_spec.lua

```diff
@@ -0,0 +1,56 @@
+local lfs = require('lfs')
+local h = require('test.functional.helpers')(after_each)
+
+local clear = h.clear
+local eq = h.eq
+local eval = h.eval
+local execute = h.execute
+local request = h.request
+
+describe('DirChanged ->', function()
+  local curdir = lfs.currentdir()
+  local dirs = {
+    curdir .. '/Xtest-functional-autocmd-dirchanged.dir1',
+    curdir .. '/Xtest-functional-autocmd-dirchanged.dir2',
+    curdir .. '/Xtest-functional-autocmd-dirchanged.dir3',
+  }
+
+  setup(function()    for _, dir in pairs(dirs) do h.mkdir(dir) end end)
+  teardown(function() for _, dir in pairs(dirs) do h.rmdir(dir) end end)
+
+  before_each(clear)
+
+  it('"autocmd DirChanged *" sets v:event for all :cd variants', function()
```

Comment:
need some tests that verify DirChanged is not published if `:{t,l}cd` fails (e.g. invalid directory)